packages/swing-store/src/transcriptStore.js

```diff
@@ -131,6 +140,14 @@ export function makeTranscriptStore(
     return transcripts;
   }
 
+  function* readVatTranscript(vatID) {
+    for (const { startPos, endPos } of sqlDumpVatSpansQuery.iterate(vatID)) {
+      for (const row of sqlDumpItemsQuery.iterate(vatID, startPos, endPos)) {
+        yield row;
```

Comment:
Good call! This method will simply be missing the historical span entries that were pruned. While this is a "full transcript" I think it's acceptable for it to yield a partial history given what we yield is both the transcript entry and its position.

I've updated the consumer (extract-transcript) to not write anything until it sees a `startVat` delivery. There was already logic to throw if the entries were not continuous (gap in positions). Before the change it would have thrown on pruned historical transcript spans. Now it will only return a full transcript of the last incarnation, but handle partially pruned transcripts of previous incarnations.